ts/Core/Axis/Color/ColorAxis.ts

```diff
@@ -401,18 +414,23 @@ class ColorAxis extends Axis implements AxisLike {
         item: ColorAxis
     ): void {
         const axis = this,
+            chart = axis.chart,
             legendItem = item.legendItem || {},
             padding = legend.padding,
             legendOptions = legend.options,
             labelOptions = axis.options.labels,
             itemDistance = pick(legendOptions.itemDistance, 10),
             horiz = axis.horiz,
             width = pick(
-                legendOptions.symbolWidth,
+                this.options.width ?
+                    relativeLength(this.options.width, chart.chartWidth) :
+                    legendOptions.symbolWidth,
```

Comment:
We now have two places in the code base that does exactly this, and one that is quite similar on like 311. Time for a refactor? What about a function like `getSize()`. 

Regarding line 311, currently it looks like this:

```
relativeLength(
                this.options.width ? this.options.width :
                    symbol.attr('width'),
                chart.chartWidth
            );
```

But I think this can be changed to work more like the two others because `symbol.attr('width'`) can never be a percentage. That allows us to create a function `getSize(width, height)` where the parameters are the default values if an option is not set.

Reply:
I refactored it, but I think, that I could overthink it, so please feel free to check it deeply.